lib/src/local_working_copy.rs

```diff
@@ -1644,6 +1777,7 @@ impl LocalWorkingCopy {
         state_path: PathBuf,
         operation_id: OperationId,
         workspace_id: WorkspaceId,
+        config: &config::Config,
```

Comment:
nit: we typically use `&UserSeettings` object.

Reply:
I'm only using config because the `Ui` object only gets initialized with a  config and it felt odd to pass user_settings into everything and then only use it by calling: `ignore_exec_bit(user_settings.config())`. I'm ok with changing it but I don't really see a reason to pass the full user settings?

Reply:
> I don't really see a reason to pass the full user settings?

I don't have a concrete plan, but we'll probably replace `config::Config` with our config module that supports layering better. Then, `config::Config` APIs might be rehosted on `UserSettings`.

Reply:
Ok, then I will move to passing UserSettings